src/common/mod.rs

```diff
@@ -25,7 +25,9 @@ pub mod error;
 mod id;
 pub(crate) mod info;
 mod options;
-mod stream;
+#[cfg_attr(not(feature = "sync"), path = "stream_async.rs")]
+#[cfg_attr(feature = "sync", path = "stream_sync.rs")]
+pub mod stream;
```

Comment:
I definitely like this, it looks nice after all! Thanks for bringing it up ahead of time